provisioner/k8s/README.md

```diff
@@ -1,3 +1,112 @@
+
+```yaml
+apiVersion: core.rukpak.io/v1alpha1
+kind: Bundle
+metadata:
+  name: my-bundle
+spec:
+  image: my-bundle-image@sha256:xyz123
+  provisionerClassName: core.rukpak.io/plain
+---
+apiVersion: core.rukpak.io/v1alpha1
+kind: BundleInstance
+metadata:
+  name: my-bundle-instance
+spec:
+  provisionerClassName: core.rukpak.io/plain
+  bundleName: my-bundle
+``` 
+### Make bundle content available but do not install it
+
+There is a natural separation between sourcing of the content and application of that content via two separate RukPak APIs, `Bundle` and `BundleInstance`. 
+A user can specify a particular `Bundle` to be available in the cluster for inspection before any application of the resources. 
+Given a `Bundle` resource named `my-bundle`, the k8s provisioner will pull down and unpack the bundle to a variety of storage backends, such as ConfigMaps or PersistentVolumes.
+
+Currently the ConfigMap backend is supported by the provisioner, and the content can be inspected by viewing the metadata ConfigMap, which holds
+information on all ConfigMaps related to this bundle. 
+
+`kubectl -n rukpak-system get cm --selector core.rukpak.io/owner-name=<bundle-name>` gets all ConfigMaps related to the `Bundle` on-cluster.
+
+Surfacing the content of a bundle in a more user-friendly way, via a plugin or additional API, is on the RukPak roadmap. 
+
+### Pivot between bundle versions 
+
+The `BundleInstance` API is meant to indicate the version of the bundle that should be active within the cluster. Given two unpacked bundles in the cluster,
+`my-bundle-v0.0.1` and `my-bundle-v0.0.2`, the `spec.bundleName` field of the related `BundleInstance` can be updated to pivot to a different version. For example by pivoting from an older version,
+to the newer version, by updating the `spec.bundleName` field the k8s provisioner will create the new bundle content on-cluster and remove the old content.  
+
+## Running locally 
+
+To experiment with the k8s provisioner locally, first setup a local cluster. 
+
+Once the cluster has been setup, take the following steps from the root of the repository:
+* Create the rukpak namespace via `kubectl create ns rukpak-system` by default
+* Run `make bin/k8s` to build the k8s provisioner binary locally
+* Apply the RukPak core CRDs via `make install-apis`
+* Run `./bin/k8s` to start the controller locally outside the cluster. 
+
+From there, creating some `Bundles` and `BundleInstance` types to see the provisioner in action. For an example bundle to use, 
+the [combo operator](https://github.com/operator-framework/combo) is a good candidate. 
+
+Create the bundle:
+```yaml
+---
+apiVersion: core.rukpak.io/v1alpha1
+kind: Bundle
+metadata:
+  name: combo-0.0.1
+spec:
+  image: quay.io/tyslaton/combo-plain-bundle:v0.0.1
+  provisionerClassName: core.rukpak.io/plain
+```
+
+Before creating the combo `BundleInstance` on-cluster, ensure that there is a namespace available that 
```

Comment:
Was there an issue with running the bundle where it didn't have the namespace available? That bundle has a manifest for one so it should come with it.

Reply:
oh nice, ok I can remove that step. I think it's leftover from the registry+v1 bundle steps